mousestyles/ultradian/__init__.py

```diff
@@ -0,0 +1,381 @@
+            elif end_index - start_index == 1:
+                time_behaviour[
+                    start_index + j *
+                    bin_count] += bin_length * end_index - start_time
+                time_behaviour[end_index + j *
+                               bin_count] += end_time % bin_length
+            else:
+                time_behaviour[
+                    start_index + j *
+                    bin_count] += bin_length * (start_index + 1) - start_time
+                time_behaviour[end_index + j *
+                               bin_count] += end_time % bin_length
+                time_behaviour[start_index + j * bin_count +
+                               1:end_index + j * bin_count] += bin_length
+
+    if feature == 'F' or feature == 'W':
+        all_feature = data.load_all_features()
+        group = all_feature[
+            ["strain", "mouse", "day", "hour", "Food", "Water"]].groupby(
+            ["strain", "mouse", "day"]).sum()
+        group = group.reset_index()
+        mouse_data = group.loc[(group['strain'] == strain) &
+                               (group['mouse'] == mouse)].copy()
+        mouse_data.loc[:, 'day'] = np.arange(len(mouse_data))
+        for i in mouse_data['day'].astype('int'):
+            if feature == 'F':
+                food_amount = float(mouse_data['Food'][mouse_data['day'] == i])
+                time_behaviour[
+                    (bin_count * i):(bin_count * (i + 1))] /= sum(
+                    time_behaviour[(bin_count * i):(bin_count * (i + 1))])
+                time_behaviour[(bin_count * i):(bin_count *
+                                                (i + 1))] *= food_amount
+            else:
+                food_amount = float(mouse_data['Water'][
+                                    mouse_data['day'] == i])
+                time_behaviour[
+                    (bin_count * i):(bin_count * (i + 1))] /= sum(
+                    time_behaviour[(bin_count * i):(bin_count * (i + 1))])
+                time_behaviour[(bin_count * i):(bin_count *
+                                                (i + 1))] *= food_amount
+    if feature == 'AS':
+        time_behaviour /= (bin_width * 60)
+
+    ts = pd.Series(time_behaviour, index=pd.date_range(
+        '01/01/2014', periods=len(time_behaviour),
+        freq=str(bin_width) + 'min'))
+    return(ts)
+
+
+def aggregate_movement(strain, mouse, bin_width):
+    """
+    Aggregate the movement data based on n-minute
+    time intervals, return a time series.
+
+    Parameters
+    ----------
+    strain: int
+        nonnegative integer indicating the strain number
+    mouse: int
+        nonnegative integer indicating the mouse number
+    bin_width: number of minutes of time interval for data aggregation
+
+    Returns
+    -------
+    ts: pandas.tseries
+        a pandas time series of length (#day)*24(hour)*60(minute)/n
+    """
+    # determine number of days
```

Comment:
Add input check for strain mouse (positive integer). Add input check for bin_width(min and max range), etc.